### PR TITLE
location_compression: safety review to init z_stream struct

### PR DESCRIPTION
**Pull request by lx18233184051**

### CODE REVIEW

**Comment by lx18233184051** [comment]
1. var_decl: Declaring variable stream_data without initializer.
6. uninit_use_in_call: Using uninitialized value stream_data.total_in when calling inflate
7. uninit_use_in_call: Using uninitialized value stream_data.total_out when calling inflate

File: modules/localization/msf/common/util/compression.cc [hunk 1]
```diff
@@ -107,6 +107,14 @@ int ZlibStrategy::ZlibUncompress(BufferStr* src, BufferStr* dst) {
   stream_data.opaque = Z_NULL;
   stream_data.avail_in = 0;
   stream_data.next_in = Z_NULL;
+  stream_data.total_in = 0ULL;
```
Comment (storypku) [inline modules/localization/msf/common/util/compression.cc:110]: Please don't try to init every member in struct one by one, which was hard to read/maintain.  Use `std::memset(...)` from <cstring> to zero-initialize the whole struct first.

Comment (lx18233184051) [reply modules/localization/msf/common/util/compression.cc:110]: ok

File: modules/localization/msf/common/util/compression.cc [hunk 2]
```diff
@@ -102,11 +102,8 @@ int ZlibStrategy::ZlibUncompress(BufferStr* src, BufferStr* dst) {
   unsigned int dst_idx = 0;
 
   /* allocate inflate state */
-  stream_data.zalloc = Z_NULL;
-  stream_data.zfree = Z_NULL;
-  stream_data.opaque = Z_NULL;
-  stream_data.avail_in = 0;
-  stream_data.next_in = Z_NULL;
+  memset(&stream_data, 0, sizeof(z_stream));
```
Comment (storypku) [inline modules/localization/msf/common/util/compression.cc:105]: Better write `std::memset`

Comment (lx18233184051) [reply modules/localization/msf/common/util/compression.cc]: ok

**Pull request merged**